File: 2021/06/23/main.go
package main

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

func reverseBetween(head *ListNode, left int, right int) *ListNode {
	new := &ListNode{}
	new.Next = head
	pre := new
	for i := 0; i < left-1; i++ {
		pre = pre.Next
	}
	curr := pre.Next
	prev := pre
	for i := 0; i <= right-left; i++ {
		prev, curr.Next, curr = curr, prev, curr.Next
	}
	pre.Next.Next = curr
	pre.Next = prev

	return new.Next
}

File: 2021/06/23/main.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseBetween(self, head: ListNode, left: int, right: int) -> ListNode:
        pre = new = ListNode(0)
        new.next = head
        for _ in range(left-1):
            pre = pre.next
        curr, prev = pre.next, None
        for _ in range(right - left + 1):
            prev, curr.next, curr = curr, prev, curr.next
        pre.next.next = curr
        pre.next = prev
        return new.next
        